lib/jbuilder.rb

```diff
@@ -252,12 +250,8 @@ def target!
 
   private
 
-  def _extract_hash_values(object, attributes)
-    attributes.each{ |key| _set_value key, object.fetch(key) }
-  end
-
-  def _extract_method_values(object, attributes)
-    attributes.each{ |key| _set_value key, object.public_send(key) }
+  def _extract_value(object, attribute)
+    object.respond_to?(attribute) ? object.public_send(attribute) : object.fetch(attribute)
```

Comment:
The risk is pretty low but still, it should rather be:

~~~ruby
object.respond_to?(:fetch) ? object.fetch(attribute) : object.public_send(attribute)
~~~

Otherwise, if a hash or hash-like object has a `:object_id`, `:hash` or any of its method as a key, the wrong value may be returned.

Could you fix this and add a test to ensure that a `Hash` with such key would be properly handled please ?

Reply:
Sorry @robin850 I just saw your comment now. Sure I'll change it 😊

Reply:
@robin850 I changed the implementation the way you suggested, however I changed respond_to?(:fetch) to respond_to?(:[]) to avoid problems when an object have a custom fetch method for other purposes.